### PR TITLE
Checking for gauge-python and getgauge version mismatch #53

### PR DESCRIPTION
**Pull request by Apoorva-GA**

### CODE REVIEW

File: start.py [hunk 1]
```diff
@@ -5,26 +5,53 @@
...
+                show_error_exit(pythonPluginVersion, getgaugeVersion)
+        else:
+            load_implementations()
+            start()
+
+def show_error_exit(pythonPluginVersion, getgaugeVersion):
+    logging.fatal('Gauge-python({}) is not compatible with getgauge({}). Please install compatible versions.\n'.format(pythonPluginVersion, getgaugeVersion))
+    exit(1)
 
+def get_version():
```
Comment (kashishm) [inline start.py]: `plugin.json` can be used to get version. https://github.com/getgauge/gauge-python/blob/babaa726bf86d5c07711e3cbc990f6405d9d2909/getgauge/impl_loader.py#L65

File: start.py [hunk 2]
```diff
@@ -5,26 +5,53 @@
...
     _init_logger()
     logging.info("Python: {}".format(platform.python_version()))
     if sys.argv[1] == "--init":
         copy_skel_files()
     else:
-        load_implementations()
-        start()
+        pythonPluginVersion = get_version()
+        getgaugeVersion = version.LooseVersion(pkg_resources.get_distribution('getgauge').version)
+        if not np.array_equal(list(map(int, pythonPluginVersion.split(".")[0:2])), getgaugeVersion.version[0:2]):
```
Comment (BugDiver) [inline start.py]: Do we really need to include a new lib (numpy) for comparing a list? I think `==` should work fine.

File: start.py [hunk 3]
```diff
@@ -5,26 +5,53 @@
...
+            show_error_exit(pythonPluginVersion, getgaugeVersion)
+        if getgaugeVersion.version.__contains__("dev") & pythonPluginVersion.find("nightly") != -1:
```
Comment (BugDiver) [inline start.py]: Any `__xxx__` is not meant to be called by user code. The keyword `in` can be used to check element existence.
Ref  http://igorsobreira.com/2010/09/16/difference-between-one-underline-and-two-underlines-in-python.html to understand diff between `_*`, `__*` and `__xx__` methods

File: start.py [hunk 4]
```diff
@@ -5,26 +5,53 @@
...
 def main():
     _init_logger()
     logging.info("Python: {}".format(platform.python_version()))
     if sys.argv[1] == "--init":
         copy_skel_files()
     else:
-        load_implementations()
-        start()
+        pythonPluginVersion = get_version()
+        getgaugeVersion = version.LooseVersion(pkg_resources.get_distribution('getgauge').version)
```
Comment (BugDiver) [inline start.py]: use snake case in naming variable and methods.

File: requirements.txt [hunk 5]
```diff
@@ -4,4 +4,5 @@ coverage>=3.7.1
 twine
 ptvsd==3.0.0
 grpcio-tools
-grpcio
\ No newline at end of file
+grpcio
+numpy
```
Comment (BugDiver) [inline requirements.txt]: If it's a runtime requirement, then need to be added in `steup.tml` as well.

**Pull request merged**